src/audio/asrc/asrc.c

```diff
@@ -876,8 +851,26 @@ static int asrc_copy(struct comp_dev *dev)
 		cd->source_frames = MIN(cd->source_frames, frames_src);
 	}
 
-	if (cd->source_frames && cd->sink_frames)
-		asrc_process(dev, source, sink);
+	if (!cd->source_frames || !cd->sink_frames)
+		return 0;
+
+	/* consumed bytes are not known at this point */
+	buffer_invalidate(source, source->stream.size);
+	cd->asrc_func(dev, &source->stream, &sink->stream, &consumed,
+		      &produced);
+	buffer_writeback(sink, produced *
+			 audio_stream_frame_bytes(&sink->stream));
```

Comment:
same, I don't get the invalidate/writeback if the intent what that the coherent API would manage this?

Reply:
ack @plbossart . It is confusing. I have renamed these now.